src/modules/status-list/status-list.service.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+} from '@nestjs/common';
+import { URL } from 'url';
+import { ConfigService } from '@nestjs/config';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { verifyCredential } from 'didkit-wasm-node';
+import { gzip } from 'pako';
+import {
+  CredentialWithStatusDto,
+  StatusList2021EntryDto,
+} from './dtos/credential-status.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { CredentialDto } from './dtos/credential.dto';
+import { VerifiableCredentialDto } from './dtos/verifiable-credential.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { StatusListCredential } from './entities/status-list-credential.entity';
+import { toHexEthrDID } from '../../common/to-hex-did';
+
+@Injectable()
+export class StatusListService {
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectRepository(StatusListCredential)
+    private readonly statusListRepository: Repository<StatusListCredential>
+  ) {}
+
+  /**
+   * Add StatusList2021Entry as `credentialStatus` parameter to given credential object.
+   * https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entry
+   *
+   * @param {CredentialDto} credential
+   * @return credential with StatusList2021Entry
+   */
+  async addStatusListEntry(
+    credential: CredentialDto
+  ): Promise<CredentialWithStatusDto> {
+    if (await this.fetchStatusList(credential.id)) {
+      throw new ConflictException('Credential already revoked');
+    }
```

Comment:
I'm not sure this behaviour is desired. I'm imaging an edge case where an issuer has called `POST /entries`, but fails to actually issue the credential for some reason (maybe their browser crashes for example). Then if they try to call `POST /entries` again for the same `credential.id`, the request would fail.
Do you think it would reasonable if instead, the behaviour was that is just populates the `credentialStatus` with the existing entry if it exists?
```suggestion
    const existingEntry = await this.fetchStatusList(credential.id);
    if (existingEntry) {
      return { ...credential, credentialStatus: existingEntry }
    }
```

Reply:
Exactly, we persisting only StatusList2021 Credential during revocation, so when revoker send signed status list credential.